### PR TITLE
config: refactor StoreConfig into PersistOptions to ensure the persistence

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #5839.

### What is changed and how does it work?

```commit-message
`StoreConfig` is needed by the scheduling service. To ensure it could be watched by the latter, this PR:

- Refactor `StoreConfig` into `PersistOptions`.
- Remove `StoreConfigManager`.
- Persist the store config after each sync.
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/config/store_config.go [hunk 1]
```diff
@@ -194,17 +195,19 @@ func (c *StoreConfig) Clone() *StoreConfig {
 type StoreConfigManager struct {
 	config atomic.Value
```
Comment (rleungx) [inline server/config/store_config.go:196]: Is there possible that we move it to persist options?

Comment (JmPotato) [reply server/config/store_config.go:196]: Good idea. Let me have a try.

**Review by lhy1024 — APPROVED** [review]

File: pkg/storage/endpoint/key_path.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ const (
 	regionLabelPath          = "region_label"
 	replicationPath          = "replication_mode"
 	customScheduleConfigPath = "scheduler_config"
+	storeConfigPath          = "store_config"
```
Comment (bufferflies) [inline pkg/storage/endpoint/key_path.go:39]: how about name `tikv_cxonfig`?

**Comment by JmPotato** [comment]
Hi, I managed to refactor the `StoreConfig` into `PersistOptions` to ensure we have a unified configuration interface, the previous review might be out of date, so please take a look again, thx.

cc @lhy1024 @bufferflies @rleungx

File: server/config/store_config.go [hunk 3]
```diff
@@ -192,99 +200,64 @@ func (c *StoreConfig) Clone() *StoreConfig {
...
 }
 
 // NewStoreConfigManager creates a new StoreConfigManager.
-func NewStoreConfigManager(client *http.Client) *StoreConfigManager {
-	schema := "http"
-	if netutil.IsEnableHTTPS(client) {
-		schema = "https"
-	}
-
+func NewStoreConfigManager(client *http.Client, persistOpts *PersistOptions) *StoreConfigManager {
```
Comment (rleungx) [inline server/config/store_config.go:208]: Do we still need the store config manager?

Comment (JmPotato) [reply server/config/store_config.go]: Removed. PTAL.

**Review by rleungx — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]

**Pull request merged**